src/LLVM_intrinsics.jl

```diff
@@ -430,10 +431,24 @@ for f in MULADD_INTRINSICS
     end
 end
 
-function fmaddsub(a::LVec{N, T}, b::LVec{N, T}, c::LVec{N, T}) where {N, T<:FloatingTypes}
-    Base.llvmcall("llvm.x86.fma.fmaddsub_pd", LVec{N, T}, (LVec{N, T}, LVec{N, T}, LVec{N, T}), a, b, c)
+
+for (t, N, T) in [("d"    , 2, Float64), ("s"    , 4, Float32),
```

Comment:
These intrinsics only exist on `x86` architectures. This function should be wrapped in an `if` statement that checks the CPU architecture, and possible also the Julia and LLVM version, to see whether the intrinsic is supported.

In case it is not, there should be a reasonably efficient generic fallback. The idea is that one can call `fmaddsub` all the time and expect a reasonable implementation.

Reply:
I genuinely cannot find when these were introduced into LLVM (I've looked at the release notes for pretty much every release since 3.0), however, they start robustly integrating AVX512 into the language at 4.0, so I'm just going to peg it at that and say FMA/AVX instructions are integrated there. I don't really know where to start with Julia versions since different installations of the same Julia version might use different LLVM versions, so I suspect that might be tricky. I'll probably steal your implementation below for the fallback, I don't really see any significant improvements.